api/base/renderers.py

```diff
@@ -12,6 +12,11 @@ def render(self, data, accepted_media_type=None, renderer_context=None):
         augmented_rendering = re.sub(r'"<esi:include src=\\"(.*?)\\"\/>"', r'<esi:include src="\1"/>', initial_rendering)
         return augmented_rendering
 
+
+class JSONRenderer(JSONAPIRenderer):
```

Comment:
Ah, I see; I had forgotten that `render` was being implemented in `JSONAPIRenderer`. I think your proposed approach makes sense: implement a `JSONRendererWithESISupport` (or a better name) then have JSONAPIRenderer subclass that. It's a minor change though: I'll leave it up to you to decide if you want to make that.